controllers/openstackdataplanerole_controller.go

```diff
@@ -106,28 +122,105 @@ func (r *OpenStackDataPlaneRoleReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, err
 	}
 
-	if len(nodes.Items) == 0 {
-		r.Log.Info("Role: ", instance.Name, "doesn't have nodes yet, requeueing")
-		return ctrl.Result{RequeueAfter: time.Second * 5}, nil
-	}
-
 	for _, node := range nodes.Items {
 		if node.Spec.Role != instance.Name {
 			err = fmt.Errorf("node %s: node.Role does not match with node.Label", node.Name)
 			return ctrl.Result{}, err
 		}
-		if !node.Status.Deployed {
-			r.Log.Info("Node: ", node.Name, "isn't deployed yet, requeueing")
-			return ctrl.Result{RequeueAfter: time.Second * 2}, nil
-		}
 	}
 
-	_, err = r.GenerateInventory(ctx, instance, nodes.Items)
+	inventoryConfigMap, err := r.GenerateInventory(ctx, instance, nodes.Items)
 	if err != nil {
 		util.LogErrorForObject(helper, err, fmt.Sprintf("Unable to generate inventory for %s", instance.Name), instance)
 		return ctrl.Result{}, err
 	}
 
+	// Always patch the instance status when exiting this function so we can
+	// persist any changes.
+	defer func() {
+		err := helper.PatchInstance(ctx, instance)
+		if err != nil {
+			r.Log.Error(_err, "PatchInstance error")
+			_err = err
+			return
+		}
+	}()
```

Comment:
please add the changes from https://github.com/openstack-k8s-operators/dataplane-operator/pull/76